rita_common/src/payment_validator/mod.rs

```diff
@@ -92,6 +97,50 @@ impl Default for PaymentValidator {
     }
 }
 
+pub fn store_payment(pmt: PaymentTx, sent_from_us: bool) {
+    let mut lock = HISTORY.write().unwrap();
+    let neighbor;
+    let data = if sent_from_us {
+        neighbor = pmt.to;
+        &mut lock.successful_transactions_sent
+    } else {
+        neighbor = pmt.from;
+        &mut lock.successful_transactions_rec
+    };
+
+    if let std::collections::hash_map::Entry::Vacant(e) = data.entry(neighbor) {
+        let mut set = HashSet::new();
+        set.insert(pmt);
+        e.insert(set);
+    } else {
+        let set = data
+            .get_mut(&neighbor)
+            .expect("This key should have an initialized set");
+        set.insert(pmt);
+    }
+}
+
+pub fn get_payment_txids(id: Identity, sent_from_us: bool) -> HashSet<PaymentTx> {
```

Comment:
I'd separate this into two function calls, it's no additonal lines actually and it's easier to see what's going on when a specific function is called.

Reply:
Fixed